### ISSUE TITLE
Support for forecast object

### ISSUE THREAD

**Issue by JHFreeman** [rstudio/dygraphs#86]
I was wondering if there is support for or a solution to using forecast and plotting the 80% and 95% prediction intervals. I tried using two dySeries

```
  hw = HoltWinters(PCYts)
  fore = forecast(hw, h = 10)
  mean = fore$mean
  lower = ts(fore$lower, start = c(2015, 8), frequency = 12)
  upper = ts(fore$upper, start = c(2015, 8), frequency = 12)
  all = cbind(xhat, PCYts, lower, mean, upper)
  dygraph(all, "PCY Enrollment") %>%
    dySeries("PCYts", label ="Actual") %>%
    dySeries("xhat", label = 'Predicted') %>%
    dySeries(c("lower.95%", "mean", "upper.95%"), label = "Forecast") %>%
    dySeries(c("lower.80%", "mean", "upper.80%"), label = NULL)
```

**Comment by thomasbierhance** [comment]
I was not able to run your code and see what the problem is. I think you can get quiet close (see code below). Wouldn't this rather be a feature request for dygraph to support more than one error band per series?

```
library(forecast)
library(ggplot2)
library(dygraphs)

fcst = stlf(ldeaths, h=12)
data = cbind(past=ldeaths,
             level80=fcst$mean,
             level95=fcst$mean,
             lwr95=fcst$lower[,"95%"],
             upr95=fcst$upper[,"95%"],
             lwr80=fcst$lower[,"80%"],
             upr80=fcst$upper[,"80%"])

dygraph(data) %>%
  dySeries("past") %>%
  dySeries(c("lwr80","level80","upr80")) %>%
  dySeries(c("lwr95","level95","upr95"))
```

**Comment by sdrumwr** [comment]
@thomasbierhance I currently have the same code above in an application. However, doing it this way means that the legend shows the same value (the mean) for both "upr80" and "upr95". This is confusing for the user and I was wondering if you have a work-around that would allow showing for example "lwr80", "upr80", "lwr95", "upr95", as well as the mean (or "prediction") in the legend. Besides the confusing legend, the user currently has to estimate the bounds of those intervals.

**Comment by thomasbierhance** [comment]
> @thomasbierhance I currently have the same code above in an application. However, doing it this way means that the legend shows the same value (the mean) for both "upr80" and "upr95".

This could be achieved using the current version of the dygraph JavaScript library. My idea would be:

1. do not draw the mean of one of the second series, by setting strokeWidth to 0
2. hide the second series from the legend by providing a custom [legendFormatter](http://dygraphs.com/options.html#legendFormatter)
3. show the intervals by providing a custom [valueFormatter](http://dygraphs.com/options.html#valueFormatter)

See this unpolished [fiddle](http://jsfiddle.net/z0f71hxk/).

However, I fear that this can not be done using the current version of the R interface to dygraphs.

**Comment by sdrumwr** [comment]
> However, I fear that this can not be done using the current version of the R interface to dygraphs.

Which is (of course) my use case! 
Thanks for the reply. I will look into the R Shiny hooks for javascript.

**Comment by thomasbierhance** [comment]
> > However, I fear that this can not be done using the current version of the R interface to dygraphs.
> 
> Which is (of course) my use case!
> Thanks for the reply. I will look into the R Shiny hooks for javascript.

I actually found a IMHO nice way to get it running with the current R interface.

See https://gist.github.com/thomasbierhance/005eac9ef2fb56d42ec0b386144b9375#file-complete_dygraph-r

This includes a custom JavaScript value formatter which is supported by the current R interface.

**Comment by amanda-hand** [comment]
> > > However, I fear that this can not be done using the current version of the R interface to dygraphs.
> > 
> > 
> > Which is (of course) my use case!
> > Thanks for the reply. I will look into the R Shiny hooks for javascript.
> 
> I actually found a IMHO nice way to get it running with the current R interface.
> 
> See https://gist.github.com/thomasbierhance/005eac9ef2fb56d42ec0b386144b9375#file-complete_dygraph-r
> 
> This includes a custom JavaScript value formatter which is supported by the current R interface.

@datenzauberai  - I think your solution is what I need for a similar problem I'm trying to solve, but the link is broken. Any chance you still have this somewhere else?

**Comment by datenzauberai** [comment]
@amanda-hand Sure, I did a small [blog post for towardsdatascience](https://towardsdatascience.com/how-to-create-better-interactive-forecast-plots-using-r-and-dygraph-29bdd7146066) about it with step-by-step explanations. In case you hit against the paywall you can find the code without explanations in this [gist](https://gist.github.com/datenzauberai/005eac9ef2fb56d42ec0b386144b9375#file-complete_dygraph-r).

**Comment by amanda-hand** [comment]
Thank you so so much! I don't see in either place a use of the custom legendFormatter. I have written a JS function to get the legend I need, my missing piece is just the dy_____ function to include it in (dyLegend won't take legendFormatter as a parameter).  This is in reference to your #2 option above, "hide the second series from the legend by providing a custom legendFormatter". I so appreciate your time, as I've spent two days trying to figure it out.

**Comment by datenzauberai** [comment]
I don’t think that this can be done easily. The R wrapper does not seem to support the legendFormatter. If I’m not mistaken the dygraphs version that ships with the wrapper is so old, that it doesn’t even contain the feature - so patching the R wrapper or injecting the option on the fly would not work.

**Comment by amanda-hand** [comment]
Well, it makes me feel better for not figuring it out!  Thanks so much again for taking the time to respond.
